<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8"/>
    <title>QConQuant parameters explained
</title>
    <meta name="viewport" content="width=device-width, initial-scale=1.0"/>
    <meta name="description" content="Project has no summmary; update build.fsx"/>
    <meta name="author" content="David Zimmer"/>

    <script src="https://code.jquery.com/jquery-1.8.0.js"></script>
    <script src="https://code.jquery.com/ui/1.8.23/jquery-ui.js"></script>
    <script src="https://netdna.bootstrapcdn.com/twitter-bootstrap/2.2.1/js/bootstrap.min.js"></script>
    <link href="https://netdna.bootstrapcdn.com/twitter-bootstrap/2.2.1/css/bootstrap-combined.min.css" rel="stylesheet"/>
    <meta http-equiv="X-UA-Compatible" content="IE=11" >
    <script src="https://cdn.plot.ly/plotly-latest.min.js"></script>
    <link type="text/css" rel="stylesheet" href="/QConQuantifier/content/style.css" /> <!--/QConQuantifier-->
    <link type="text/css" rel="stylesheet" href="/QConQuantifier/styles/CSBStyles.css" /> <!--/QConQuantifier-->
    <script type="text/javascript" src="/QConQuantifier/content/tips.js"></script> <!--/QConQuantifier-->
    <!-- HTML5 shim, for IE6-8 support of HTML5 elements -->
    <!--[if lt IE 9]>
      <script src="https://oss.maxcdn.com/html5shiv/3.7.2/html5shiv.min.js"></script>
    <![endif]-->
  </head>
  <body>
    <div class="container">
      <div class="masthead">
        <ul class="nav nav-pills pull-right">
          <li><a href="http://fsharp.org">fsharp.org</a></li>
          <li><a href="https://github.com/ZimmerD/QConQuantifier">github page</a></li>
        </ul>
        <h3 class="muted"><a href="/QConQuantifier/index.html">QConQuantifier</a></h3>
      </div>
      <hr />
      <div class="row">
        <div class="span9" id="main">
          
<h1><a name="QConQuant-parameters-explained" class="anchor" href="#QConQuant-parameters-explained">QConQuant parameters explained</a></h1>
<p>We have already encountered an of the parameters needed to run this tool going through the example on the landing page. This tutorial aims to
go through this monstrosity one by one and get a little more into detail.</p>
<table class="pre"><tr><td class="lines"><pre class="fssnip"><span class="l"> 1: </span>
<span class="l"> 2: </span>
<span class="l"> 3: </span>
<span class="l"> 4: </span>
<span class="l"> 5: </span>
<span class="l"> 6: </span>
<span class="l"> 7: </span>
<span class="l"> 8: </span>
<span class="l"> 9: </span>
<span class="l">10: </span>
<span class="l">11: </span>
<span class="l">12: </span>
<span class="l">13: </span>
<span class="l">14: </span>
<span class="l">15: </span>
<span class="l">16: </span>
<span class="l">17: </span>
<span class="l">18: </span>
<span class="l">19: </span>
<span class="l">20: </span>
<span class="l">21: </span>
<span class="l">22: </span>
<span class="l">23: </span>
<span class="l">24: </span>
<span class="l">25: </span>
<span class="l">26: </span>
<span class="l">27: </span>
<span class="l">28: </span>
<span class="l">29: </span>
<span class="l">30: </span>
<span class="l">31: </span>
<span class="l">32: </span>
<span class="l">33: </span>
<span class="l">34: </span>
<span class="l">35: </span>
<span class="l">36: </span>
<span class="l">37: </span>
<span class="l">38: </span>
<span class="l">39: </span>
<span class="l">40: </span>
<span class="l">41: </span>
<span class="l">42: </span>
<span class="l">43: </span>
<span class="l">44: </span>
<span class="l">45: </span>
<span class="l">46: </span>
<span class="l">47: </span>
<span class="l">48: </span>
<span class="l">49: </span>
<span class="l">50: </span>
<span class="l">51: </span>
<span class="l">52: </span>
<span class="l">53: </span>
<span class="l">54: </span>
<span class="l">55: </span>
<span class="l">56: </span>
<span class="l">57: </span>
<span class="l">58: </span>
<span class="l">59: </span>
<span class="l">60: </span>
<span class="l">61: </span>
<span class="l">62: </span>
<span class="l">63: </span>
<span class="l">64: </span>
<span class="l">65: </span>
<span class="l">66: </span>
</pre></td>
<td class="snippet"><pre class="fssnip highlighted"><code lang="fsharp"><span class="k">let</span> <span onmouseout="hideTip(event, 'fs11', 13)" onmouseover="showTip(event, 'fs11', 13)" class="id">qConQuantifierParams</span> <span class="o">=</span> 
    <span class="pn">{</span>
    <span class="c">// This will be the name of the peptide data base that is created upon application launch.</span>
    <span class="id">Name</span>                            <span class="o">=</span> <span class="s">&quot;ChlamyTruncDB&quot;</span>
    <span class="c">// This is the directory were the data base will be located. Whenever</span>
    <span class="c">// the application is started the tool looks for data bases in this directory and retrieves their</span>
    <span class="c">// creation parameters. If they are identical to the once in use, the data base will be reused.</span>
    <span class="c">// Note: this also means that this tool will not delete a data base ;) </span>
    <span class="id">DbFolder</span>                        <span class="o">=</span>  <span class="s">@&quot;C:\Users\david\Source\Repos\netCoreRepos\QConcatifier_Samples\db&quot;</span>
    <span class="c">// This collection of file paths should point to your qConCAT fastas of interest. It is possible to use more than one QconCAT .fasta</span>
    <span class="c">// in one experiment</span>
    <span class="id">QConCatFastaPaths</span>               <span class="o">=</span> <span class="pn">[</span><span class="s">@&quot;C:\Users\david\Source\Repos\netCoreRepos\QConcatifier_Samples\fasta\PS QconCAT.fasta&quot;</span><span class="pn">]</span>
    <span class="c">// The QConCATifier uses the method of target-decoy competition distinct true postive peptide spectrum matches (PSMs) from false positive ones. This data base</span>
    <span class="c">// serves as a codon-unbiased target and decoy source. </span>
    <span class="id">OrganismFastaPath</span>               <span class="o">=</span>  <span class="s">@&quot;C:\Users\david\Source\Repos\netCoreRepos\QConcatifier_Samples\fasta\Chlamy_JGI5_trunc.fasta&quot;</span>
    <span class="c">// Since .fasta headers are a the wild west of bioinformatics you can specify a regex pattern to parse your protein identifier - if left blank the complete</span>
    <span class="c">// fasta Header will be used in data base creation</span>
    <span class="id">ParseProteinIDRegexPattern</span>      <span class="o">=</span> <span class="s">&quot;id&quot;</span>
    <span class="c">// The protease used for in silico digestion of QConCAT and organism proteins. Note: Should match to the protease used to design the QConCAT. </span>
    <span onmouseout="hideTip(event, 'fs12', 14)" onmouseover="showTip(event, 'fs12', 14)" class="id">Protease</span>                        <span class="o">=</span> <span onmouseout="hideTip(event, 'fs12', 15)" onmouseover="showTip(event, 'fs12', 15)" class="m">Protease</span><span class="pn">.</span><span class="id">Trypsin</span>
    <span class="c">// Minimum number of missed cleavages to scan for.</span>
    <span class="id">MinMissedCleavages</span>              <span class="o">=</span> <span class="n">0</span>
    <span class="c">// Minimum numer of missed cleavages to scan for.</span>
    <span class="id">MaxMissedCleavages</span>              <span class="o">=</span> <span class="n">1</span>
    <span class="c">// Maximum peptide mass allowed in peptide data base.</span>
    <span class="id">MaxMass</span>                         <span class="o">=</span> <span class="n">15000.</span>
    <span class="c">// Minimum peptide length allowed in peptide data base.</span>
    <span class="id">MinPepLength</span>                    <span class="o">=</span> <span class="n">5</span>
    <span class="c">// Maximum peptide length  allowed in peptide data base.</span>
    <span class="id">MaxPepLength</span>                    <span class="o">=</span> <span class="n">50</span>
    <span class="c">// Isotopic modification used to lable the experiment.</span>
    <span onmouseout="hideTip(event, 'fs13', 16)" onmouseover="showTip(event, 'fs13', 16)" class="id">IsotopicMod</span>                     <span class="o">=</span> <span class="pn">[</span><span onmouseout="hideTip(event, 'fs13', 17)" onmouseover="showTip(event, 'fs13', 17)" class="m">IsotopicMod</span><span class="pn">.</span><span class="id">N15</span><span class="pn">]</span>
    <span class="c">// Selected mass mode. High accuracy mass spectrometers deliver a resolution that makes it possible to resolve monoisotopic peaks. </span>
    <span class="c">// If an old instrument is used, the mass mode can be switched to average mass. This is then needed to simulate low resolution theoretical spectra during peptide spectrum</span>
    <span class="c">// matching.</span>
    <span onmouseout="hideTip(event, 'fs14', 18)" onmouseover="showTip(event, 'fs14', 18)" class="id">MassMode</span>                        <span class="o">=</span> <span onmouseout="hideTip(event, 'fs14', 19)" onmouseover="showTip(event, 'fs14', 19)" class="m">MassMode</span><span class="pn">.</span><span class="id">Monoisotopic</span>
    <span class="c">// If the experimental design led to a permanent modification of a certain amino acid (e.g. carbamidomethylation), this has to be accounted for when computing</span>
    <span class="c">// in silico spectra and retrieving look ups. Here you can specify you modifications.</span>
    <span class="id">FixedMods</span>                       <span class="o">=</span> <span class="pn">[</span><span class="pn">]</span>            
    <span class="c">// Sometimes the experimental design leads to a variable modification of a certain amino acid (e.g. methionin oxidation) or one is </span>
    <span class="c">// interested in post translational modification (e.g. phosphorylations). Here you can specify your variable modifications.</span>
    <span class="id">VariableMods</span>                    <span class="o">=</span> <span class="pn">[</span><span class="pn">]</span>
    <span class="c">// This threshold determines how many variable modifications the user allows to be used for one amino acid. It is not recommended to rise this number to much</span>
    <span class="c">// because it is likely to result in a combinatorial explosion.</span>
    <span class="id">VarModThreshold</span>                 <span class="o">=</span> <span class="n">3</span>
    <span class="c">// Expected minimum charge, important for calculating theoretically possible mzs of qConCATs</span>
    <span class="id">ExpectedMinCharge</span>               <span class="o">=</span> <span class="n">2</span>
    <span class="c">// Expected maximum charge, important for calculating theoretically possible mzs of qConCATs</span>
    <span class="id">ExpectedMaxCharge</span>               <span class="o">=</span> <span class="n">3</span>
    <span class="c">// When a ms2 is recorded, this decision is based on a ion observed in a ms1 scan. The charge and the m/z value of this precursor ion</span>
    <span class="c">// determine the search space. This search window is determined by the theoretical mass +/- x. The x is typically calculated as parts per million</span>
    <span class="c">// of the mass.</span>
    <span class="id">LookUpPPM</span>                       <span class="o">=</span> <span class="n">30.</span>
    <span class="c">// Determines the scan range when calculating in silico spectra</span>
    <span class="id">ScanRange</span>                       <span class="o">=</span> <span class="n">100.0</span><span class="pn">,</span><span class="n">1600.0</span>
    <span class="c">// Determines if the target-decoy competition is used (PepValue) or if the spectra should be filtered solely on the sequest score. </span>
    <span onmouseout="hideTip(event, 'fs15', 20)" onmouseover="showTip(event, 'fs15', 20)" class="id">PSMThreshold</span>                    <span class="o">=</span> <span onmouseout="hideTip(event, 'fs15', 21)" onmouseover="showTip(event, 'fs15', 21)" class="rt">PSMThreshold</span><span class="pn">.</span><span class="id">PepValue</span> <span class="n">0.05</span>
    <span class="c">// Determines the scan time window in minutes when extracting a XIC.</span>
    <span class="id">ScanTimeWindow</span>                  <span class="o">=</span> <span class="n">2.5</span>
    <span class="c">// Determines the m/z window in dalton when extracting a XIC.</span>
    <span class="id">MzWindow_Da</span>                     <span class="o">=</span> <span class="n">0.07</span>
    <span class="c">// Determines the nterminal series considered when calculating in silico spectra</span>
    <span onmouseout="hideTip(event, 'fs16', 22)" onmouseover="showTip(event, 'fs16', 22)" class="id">NTerminalSeries</span>                 <span class="o">=</span> <span onmouseout="hideTip(event, 'fs16', 23)" onmouseover="showTip(event, 'fs16', 23)" class="m">NTerminalSeries</span><span class="pn">.</span><span onmouseout="hideTip(event, 'fs17', 24)" onmouseover="showTip(event, 'fs17', 24)" class="id">B</span>
    <span class="c">// Determines the cterminal series considered when calculating in silico spectra</span>
    <span onmouseout="hideTip(event, 'fs18', 25)" onmouseover="showTip(event, 'fs18', 25)" class="id">CTerminalSeries</span>                 <span class="o">=</span> <span onmouseout="hideTip(event, 'fs18', 26)" onmouseover="showTip(event, 'fs18', 26)" class="m">CTerminalSeries</span><span class="pn">.</span><span onmouseout="hideTip(event, 'fs19', 27)" onmouseover="showTip(event, 'fs19', 27)" class="id">Y</span>
    <span class="pn">}</span> 
</code></pre></td>
</tr>
</table>

<div class="tip" id="fs1">namespace System</div>
<div class="tip" id="fs2">namespace System.IO</div>
<div class="tip" id="fs3">namespace Deedle</div>
<div class="tip" id="fs4">namespace FSharpAux</div>
<div class="tip" id="fs5">namespace BioFSharp</div>
<div class="tip" id="fs6">namespace BioFSharp.Mz</div>
<div class="tip" id="fs7">namespace QConQuantifier</div>
<div class="tip" id="fs8">module Parameters<br /><br />from QConQuantifier</div>
<div class="tip" id="fs9">module Domain<br /><br />from QConQuantifier.Parameters</div>
<div class="tip" id="fs10">module DTO<br /><br />from QConQuantifier.Parameters</div>
<div class="tip" id="fs11">val qConQuantifierParams : QConQuantifierParams</div>
<div class="tip" id="fs12">Multiple items<br />module Protease<br /><br />from QConQuantifier.Parameters.DTO<br /><br />--------------------<br />type Protease = | Trypsin</div>
<div class="tip" id="fs13">Multiple items<br />module IsotopicMod<br /><br />from QConQuantifier.Parameters.DTO<br /><br />--------------------<br />type IsotopicMod =<br />&#160;&#160;| N15<br />&#160;&#160;| C13</div>
<div class="tip" id="fs14">Multiple items<br />module MassMode<br /><br />from QConQuantifier.Parameters.DTO<br /><br />--------------------<br />type MassMode = SearchDB.MassMode</div>
<div class="tip" id="fs15">type PSMThreshold =<br />&#160;&#160;| PepValue of float<br />&#160;&#160;| SequestScore of float</div>
<div class="tip" id="fs16">Multiple items<br />module NTerminalSeries<br /><br />from QConQuantifier.Parameters.DTO<br /><br />--------------------<br />type NTerminalSeries =<br />&#160;&#160;| A<br />&#160;&#160;| B<br />&#160;&#160;| C<br />&#160;&#160;| AB<br />&#160;&#160;| AC<br />&#160;&#160;| BC<br />&#160;&#160;| ABC</div>
<div class="tip" id="fs17">union case NTerminalSeries.B: NTerminalSeries</div>
<div class="tip" id="fs18">Multiple items<br />module CTerminalSeries<br /><br />from QConQuantifier.Parameters.DTO<br /><br />--------------------<br />type CTerminalSeries =<br />&#160;&#160;| X<br />&#160;&#160;| Y<br />&#160;&#160;| Z<br />&#160;&#160;| XY<br />&#160;&#160;| XZ<br />&#160;&#160;| YZ<br />&#160;&#160;| XYZ</div>
<div class="tip" id="fs19">union case CTerminalSeries.Y: CTerminalSeries</div>

        </div>
        <div class="span3">
          <img src="/QConQuantifier/img/logo.png" alt="F# Project" style="width:150px;margin:10px" />  
          <ul class="nav nav-list" id="menu" style="margin-top: 20px;">
              <li class="nav-header">QConQuantifier</li>
              <li><a href="/QConQuantifier/index.html">Home page</a></li>
              <li class="divider"></li>
              <li><a href="https://github.com/ZimmerD/QConQuantifier">Source Code on GitHub</a></li>
              <li><a href="/QConQuantifier/license.html">License</a></li>
              <li><a href="/QConQuantifier/release-notes.html">Release Notes</a></li>

              <li class="nav-header">Getting started</li>
              <li><a href="/QConQuantifier/WhyQConCats.html">Why QConCATs?</a></li>
              <li><a href="/QConQuantifier/Parameters.html">Parameters explained</a></li>
              <li class="nav-header">Documentation</li>
              <li><a href="/QConQuantifier/reference/index.html">API Reference</a></li>
          </ul>
        </div>
      </div>
    </div>
    <a href="https://github.com/ZimmerD/QConQuantifier"><img style="position: absolute; top: 0; right: 0; border: 0;" src="https://s3.amazonaws.com/github/ribbons/forkme_right_gray_6d6d6d.png" alt="Fork me on GitHub"/></a>
  </body>
  </html>
